src/main/java/dogebot/TaskList.java

```diff
@@ -0,0 +1,207 @@
+        tasks.add(temp);
+        System.out.println("\t" + temp.toString());
+        updateTasksCounter();
+    }
+
+    /**
+     * Creates new Deadline task.
+     *
+     * @param words User command for deadline task.
+     * @throws DogeBotException If input is empty.
+     */
+    public static void deadline(String words) throws DogeBotException {
+        if (words.isBlank()) {
+            throw new DogeBotException("Oops ! The description of a deadline cannot be empty :(");
+        }
+
+        int split = words.indexOf("/by");
+        // substring w/o the spaces
+        String taskDescription = words.substring(0, split - 1);
+        String taskDeadline = words.substring(split + 4, words.length());
+
+        System.out.println("Mama mia ! I've just added this task:");
+        Task temp = new Deadline(taskDescription, taskDeadline, false);
+        tasks.add(temp);
+        System.out.println("\t" + temp.toString());
+        updateTasksCounter();
+    }
+
+    /**
+     * Creates new Event task.
+     *
+     * @param words User command for event task.
+     * @throws DogeBotException If input is empty.
+     */
+    public static void event(String words) throws DogeBotException {
+        if (words.isBlank()) {
+            throw new DogeBotException("Oops ! The description of an event cannot be empty :(");
+        }
+
+        // substring w/o the spaces
+        int startSplit = words.indexOf("/from");
+        String taskDescription = words.substring(0, startSplit - 1);
+        int endSplit = words.indexOf("/to", startSplit + 1); // find "/" after startSplit index
+        String start = words.substring(startSplit + 6, endSplit - 1);
+        String end = words.substring(endSplit + 4, words.length());
+
+        System.out.println("Mama mia ! I've just added this task:");
+        Task temp = new Event(taskDescription, start, end, false);
+        tasks.add(temp);
+        System.out.println("\t" + temp.toString());
+        updateTasksCounter();
+    }
+
+    /**
+     * Deletes a task with its index as input.
+     *
+     * @param index Index of task to delete.
+     * @throws DogeBotException If task list is empty.
+     */
+    public static void delete(int index) throws DogeBotException {
+        if (tasks.size() == 0) {
+            throw new DogeBotException("Oops ! There's no tasks in your list to delete :O");
+        }
+
+        Task curr = tasks.get(index);
+        System.out.println("Got it~ This task has been removed:");
+        System.out.println("\t" + curr.toString());
+        tasks.remove(index);
+        updateTasksCounter();
+    }
+
+    /**
+     * Retrieves tasks with the input keyword.
+     *
+     * @param s Input keyword.
+     * @throws DogeBotException If input keyword is empty.
+     */
+    public static void find(String s) throws DogeBotException {
+        if (s.isBlank()) {
+            throw new DogeBotException("Oops ! The keyword cannot be empty :(");
```

Comment:
Would it be better if exceptions are created based on the events eg. KeywordNotFoundException and these specific exceptions extends from DogeBotException?